### PR TITLE
Add OpenAPI Specification, Sanitations and Generated Client

### PR DESCRIPTION
**Pull request by SanduniU**
**Purpose**

Adding OpenAI OpenAPI specification, Adding sanitations and generating the client, types, and utilities for the API

**Approach**

- Added the OpenAI OpenAPI spec to the project.
- Did necessary sanitations on the OpenAPI Specification and documented the sanitations in sanitations.md
- Generated client code based on the new API specification.
- Generated type definitions for the updated API endpoints.
- Generated utility functions to support the new client and type definitions.
- Changed .gitignore
- Added Dependencies.toml

### CODE REVIEW

File: ballerina/tests/test.bal
Comment (NipunaRanasinghe) [inline ballerina/tests/test.bal:1]: @SanduniU IMO we can remove these test skeletons from this PR and, send the completed test suite in a separate PR.

Comment (SanduniU) [reply ballerina/tests/test.bal:1]: Test directory was deleted from this branch

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM
